code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt

```diff
@@ -730,26 +896,47 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
         api.dispatch(responseEvent)
     }
 
+    /**
+     * Updates this instance of [AnalyticsState] with the shared state data for the extensions
+     * listed in [dependencies] at the version of the given {@code event}.
+     *
+     * @param event the triggering event
+     * @param dependencies list of extension names to retrieve state data
+     *
+     * @see AnalyticsState.update
+     */
     private fun updateAnalyticsState(event: Event, dependencies: List<String>) {
         val map = dependencies.associateWith { extensionName ->
             api.getSharedState(extensionName, event, true, SharedStateResolution.ANY)?.value
         }
         analyticsState.update(map)
     }
 
+    /**
+     * Processes track requests by calling [track] if the given [data] contains
+     * track action or track state data.
+     *
+     * @param event event which observed the tracking request
+     * @param data analytics tracking data
+     */
     private fun handleTrackRequest(event: Event, data: Map<String, Any?>) {
         if (data.isEmpty()) {
-            Log.debug(AnalyticsConstants.LOG_TAG, CLASS_NAME, "track - event data is null or empty.")
+            Log.debug(AnalyticsConstants.LOG_TAG, CLASS_NAME, "handleTrackRequest - event data is null or empty.")
             return
         }
-        if (data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.TRACK_ACTION) ||
-            data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.TRACK_STATE) ||
-            data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.CONTEXT_DATA)
-        ) {
+        if (isTrackActionOrTrackStateEvent(data)) {
```

Comment:
Call the existing `isTrackActionOrTrackStateEvent()` which performs the same check.